gui/default/syncthing/core/syncthingController.js

```diff
@@ -1508,6 +1510,60 @@ angular.module('syncthing.core')
             });
         };
 
+        $scope.statusOtherDevices = function () {
+            var unknownArray = [], unusedArray = [], pausedArray = [], insyncArray = [], disconnectedArray = [], defaultArray =[];
+
+            // loop through all devices
+            var devices = $scope.otherDevices();
+            var deviceLength = devices.length;
+            for (var i = 0; i < deviceLength; i++) {
+                var status = $scope.deviceStatus({
+                    deviceID:$scope.devices[i].deviceID
+                });
+
+                switch (status){
+                    case 'unused':
+                        unusedArray.push(devices[i]);
+                        break;
+
+                    case 'unknown':
+                        unknownArray.push(devices[i]);
+                        break;
+
+                    case 'paused':
+                        pausedArray.push(devices[i]);
+                        break;
+
+                    case 'insync' || 'syncing':
+                        insyncArray.push(devices[i]);
+                        break;
+
+                    case 'disconnected':
+                        disconnectedArray.push(devices[i]);
+                        break;
+
+                    default:
+                        defaultArray.push(devices[i]);
+                }
+            };
+
+            return unknownArray.concat(insyncArray, disconnectedArray, pausedArray, unusedArray, defaultArray);
+        };
+
+        $scope.guiShowOtherDevices = function () {
+            if($scope.statusSort == 'undefined'){
```

Comment:
I don't understand what's going on in here, maybe I missed something in a previous discussion. If `statusSort` is `undefined` (which is better checked as just `$scope.statusSort === undefined` without the string conversion, btw), we return a future-scheduled function that checks again the value of `statusSort` (which we already determined is undefined), and ... what? :)

Reply:
I honestly did not remember why I coded like that, it makes double check may be. Of course, I changed it. Thanks.